h5py/_errors.pyx

```diff
@@ -120,7 +120,7 @@ cdef int set_exception() except -1:
 
     err.n = -1
 
-    if H5Ewalk(<hid_t>H5E_DEFAULT, H5E_WALK_UPWARD, walk_cb, &err) < 0:
+    if H5Ewalk(<hid_t>H5E_DEFAULT, H5E_WALK_UPWARD, &walk_cb, &err) < 0:
```

Comment:
Has this just been super broken?

Reply:
No, it is not broken. I just realized that an operator `&` applied to a function pointer has [no effect](https://stackoverflow.com/questions/6893285/why-do-function-pointer-definitions-work-with-any-number-of-ampersands-or-as). So that part would not be necessary.